src/app/services/covid-data-api.service.ts

```diff
@@ -32,13 +32,13 @@ export class CovidDataApiService {
    * @param {Observable<CovidDataPoint>} method The method to be called
    */
   private getResultFromOneApiFor(
-    country?: Country
+    fetchCommand: ApiFetchCommand
   ): Observable<CovidDataPoint> {
     let errorCount = 0; // keep track of number or errored out requests for later
 
     return race( // only need result from one of the apis
       this.apiServices.map<Observable<CovidDataPoint>>((api, i, apis) =>
-        new GetLatestDataCommand(api, country).call().pipe(
+        fetchCommand.call(api).pipe(
           catchError((err) => {
```

Comment:
So at this point, instead of using command pattern, wouldn't it be simpler to just add call to each of the apis?
Then you'd have e.g.
```
api.call(callback, errorCallback)
```

Reply:
maybe not, since the motivation behind the `getResultFromOneApiFor` method is to have a generic way to cycle through each subservice and calling the same method on each one. 

So as long as you know what subservice method you're calling and what arguments to feed it then `getResultFromOneApiFor` would be able to do its job.

Had introduced the command pattern because the api in this case has more than one method that could be called, which is decided in the `GetLatestDataCommand`'s call.

Reply:
We don't need to have 2 api methods though. Instead of having one for global and one for local data, we can just have one call method which returns the correct data given the input from the region service?

Reply:
Actually, for the summary api we don't even need to distinguish between global and per-country endpoint, we can fetch all the country data with one call.